inference/enterprise/workflows/entities/steps.py

```diff
@@ -676,15 +855,26 @@ def validate_field_binding(self, field_name: str, value: Any) -> None:
 
 
 class BarcodeDetection(BaseModel, StepInterface):
+    model_config = ConfigDict(
+        json_schema_extra={
+            "description": "This block represents inference from barcode detection.",
+            "docs": None,
+        }
+    )
     type: Literal["BarcodeDetection"]
-    name: str
-    image: Union[str, List[str]]
+    name: str = Field(description="Unique name of step in workflows")
+    image: Union[InferenceImageSelector, OutputStepImageSelector] = Field(
```

Comment:
Do we have a heuristic for which models are most likely to want to be used on the original image vs a cropped/transformed image? I was trying to come up with a heuristic to wire ObjectDetectionModel to input and BarcodeDetection to a crop output but not sure there is a good one.

Reply:
not sure if this is quasi optimal heuristic.
What I would do in UI (assuming we are talking about this use case) is suggesting wiring to input image once crop is not chosen in the workflow scratch - and in opposite case I would suggest to connect to crop.
It is hard to come up with good heuristic for that - I am not sure about details of `BarcodeDetection` - from what I saw in examples it detects barcode, not only reads it